src/pybind/mgr/cephadm/module.py

```diff
@@ -2265,10 +2262,26 @@ def _apply_all_services(self):
                     spec.service_name(), spec, e))
         return r
 
-    def _refresh_configs(self):
+    def _check_daemons(self):
+        # get monmap mtime so we can refresh configs when mons change
+        monmap = self.get('mon_map')
+        last_monmap: Optional[datetime.datetime] = datetime.datetime.strptime(
+            monmap['modified'], CEPH_DATEFMT)
+        if last_monmap and last_monmap > datetime.datetime.utcnow():
+            last_monmap = None   # just in case clocks are skewed
+
         daemons = self.cache.get_daemons()
         grafanas = []  # type: List[orchestrator.DaemonDescription]
         for dd in daemons:
+            # orphan?
+            if dd.service_name() not in self.spec_store.specs and \
```

Comment:
In a post upgrade scenario, this would remove services, right?

Reply:
yes.  if you have a cluster running octopus dev code, make sure you create the specs before upgrading.

Reply:
This is probably something we can automate in a nautilus->octopus upgrade.

It shouldn't be terribly hard to translate running services to valid ServiceSpecs.

Reply:
no such thing as a nautilus->octopus upgrade with cephadm...

we do need to adjust the adoption workflow, though.  i'll update the docs.

Reply:
well, there is going to be a deepsea to cephadm workflow needed for octopus